scripts/plots/main_plot_webview.js

```diff
@@ -1,42 +1,233 @@
+            ? decodeURIComponent(src).replace(/data:image\/svg\+xml,/, '')
+            : null
+        const png = src.includes('image/png')
+            ? src.replace(/data:image\/png;base64,/, '')
+            : null
+        const gif = src.includes('image/gif')
+            ? src.replace(/data:image\/gif;base64,/, '')
+            : null
+
+        postMessageToHost(SAVE_PLOT_MESSAGE_TYPE, { svg, png, gif, index })
+    }
+}
+
+function handlePlotCopyRequest() {
+    const plot = document.querySelector('svg') || getPlotElement()
+    const isSvg = document.querySelector('svg') !== null
+
+    const width = plot.offsetWidth
+    const height = plot.offsetHeight
+
+    if (isSvg) {
+        const canvas = document.createElement('canvas')
+        const ctx = canvas.getContext('2d')
+
+        const image = new Image()
+        const data = (new XMLSerializer()).serializeToString(plot)
+        const blob = new Blob([data], { type: 'image/svg+xml;charset=utf-8' })
+        const url = window.URL.createObjectURL(blob)
+
+        image.onload = () => {
+            canvas.width = image.naturalWidth
+            canvas.height = image.naturalHeight
+            ctx.drawImage(image, 0, 0)
+            window.URL.revokeObjectURL(url)
+
+            canvas.toBlob((blob) => {
+                navigator.clipboard.write([
+                    new ClipboardItem({
+                        [blob.type]: blob
+                    })
+                ])
+            })
+        }
+        image.src = url
+    } else {
+        html2canvas(plot, { height, width }).then(
+            (canvas) => {
+                canvas.toBlob((blob) => {
+                    navigator.clipboard.write([
+                        new ClipboardItem({
+                            [blob.type]: blob
+                        })
+                    ])
+                })
+            },
+            (reason) => {
+                console.error(new Error(reason))
+            }
+        )
+    }
+}
+
+/**
+ *
+ * @param {number} index
+ * @param { REQUEST_ZOOM_IN_PLOT_TYPE | REQUEST_ZOOM_RESET_PLOT_TYPE | REQUEST_ZOOM_OUT_PLOT_TYPE } reqType
+ */
+function handlePlotZoomRequest(index, reqType) {
+    switch (reqType) {
+    case REQUEST_ZOOM_IN_PLOT_TYPE:
+        g_svgPanZoomController.zoomIn()
+        break
+    case REQUEST_ZOOM_RESET_PLOT_TYPE:
+        g_svgPanZoomController.reset()
+        break
+    case REQUEST_ZOOM_OUT_PLOT_TYPE:
+        g_svgPanZoomController.zoomOut()
+        break
+    default:
+        console.error(new Error('Unknown plot request!'))
+    }
+}
+
+/**
+ * Remove Plotly builtin export button; it's nonfunctional in VSCode and can confuse users.
+ */
+function removePlotlyBuiltinExport() {
+    if (isPlotly()) {
+        document.querySelector(
+            '[data-title="Download plot as a png"]'
+        ).style.display = 'none'
+    }
+}
+
+function initSvgPanZoom() {
+    const svgTag = document.querySelector('svg')
```

Comment:
This doesn't work for non-svg plots. I haven't really looked at svg-pan-zoom, but would expect it to work fine with other `img`s as well, so maybe 
```suggestion
    const svgTag = getPlotElement()
```
would work fine? If not, then this a `isSVG` guard.

Reply:
* Sometimes the `plot_element` is a `div` wrapping the actual plot. 
* All `Plots` backends are displayed as SVG, `Plotly` is ignored because it has its own controllers.
* `Vega` plots were displayed as `canvas` but I had reconfigured it to be displayed as `svg`.

> but would expect it to work fine with other imgs as well, so maybe

It only supports `<svg>` tags, even `img` with svg `src` isn't supported, but it isn't a problem as explained above.